src/main/java/net/snowflake/client/jdbc/SnowflakeFileTransferAgent.java

```diff
@@ -1662,14 +1663,35 @@ public InputStream downloadStream(String fileName) throws SnowflakeSQLException
 
     remoteLocation remoteLocation = extractLocationAndPath(stageInfo.getLocation());
 
-    String stageFilePath = fileName;
+    // when downloading files as stream there should be only one file in source files
+    String sourceLocation =
+        sourceFiles.stream()
+            .findFirst()
+            .orElseThrow(
+                () ->
+                    new SnowflakeSQLException(
+                        queryID,
+                        SqlState.NO_DATA,
+                        ErrorCode.FILE_NOT_FOUND.getMessageCode(),
+                        session,
+                        "File not found: " + fileName));
+
+    String stageFilePath;
+    if (fileName.equals(sourceLocation)) {
+      stageFilePath = fileName;
+    } else {
+      // filename may be different from source location e.g. in git repositories
+      logger.debug("Changing file to download location from {} to {}", fileName, sourceLocation);
```

Comment:
The assignments in both branches of this if are equivalent, so we could move them out of the if, leaving just `if (!fileName.equals(sourceLocation)){\*log*\}`

Reply:
Changed